app/controllers/categories_controller.rb

```diff
@@ -2,14 +2,25 @@
 
 class CategoriesController < ApplicationController
   include CollectionPageSetupConcern
+  include CategoriesHelper
   include Shared
   before_action :set_category, only: %i[show edit update destroy]
+  respond_to :json, only: [:index]
 
   # GET /categories
   # GET /categories.json
 
   def index
     page_collection('@categories', 'category')
+    respond_to do |format|
+      format.json do
+        render json: {
+          data: categories_or_moods_props(@categories),
+          lastPage: @categories.last_page?
+        }
+      end
+      format.html
```

Comment:
Do we actually need `format.html` here and in other similar places? I'm thinking we don't cause we're using json.